sympy/calculus/util.py

```diff
@@ -1627,5 +1481,103 @@ def union(self, other):
             return AccumBounds(other.min, Max(self.max, other.max))
 
 
+@dispatch(AccumulationBounds, AccumulationBounds) # type: ignore # noqa:F811
+def _eval_is_le(lhs, rhs): # noqa:F811
+    if lhs.max <= rhs.min:
+        return True
+    if lhs.min > rhs.max:
+        return False
+
+@dispatch(AccumulationBounds, Basic) # type: ignore # noqa:F811
+def _eval_is_le(lhs, rhs): # noqa: F811
```

Comment:
I don't see why we need to allow `Basic` rather than `Expr` for inequalities.

Using `<` currently gives a TypeError in that situation:
```julia
In [1]: Expr() < Basic()                                                                                                                       
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-4d9c8665d934> in <module>
----> 1 Expr() < Basic()

TypeError: '<' not supported between instances of 'Expr' and 'Basic'

In [2]: Expr() < Expr()                                                                                                                        
Out[2]: Expr() < Expr()
```

Reply:
Actually I think it's important that `is_le` etc only work for `Expr`. We should ensure that there are no handlers for `Basic` so that `is_le(Basic(), Basic())` gives a `TypeError`.

Reply:
Ok, I'll put that in and add a unit test for is_le(Basic(), Basic()) throwing a TypeError